src/lib.rs

```diff
@@ -69,6 +69,12 @@ pub trait Target {
     /// register's value.
     fn read_registers(&mut self, push_reg: impl FnMut(&[u8]));
 
+    /// Write the target's registers.
+    ///
+    /// The registers are provided in the order specified in the target's
+    /// `target.xml`.
+    fn write_registers(&mut self, regs: impl FnMut() -> Option<Self::Usize>);
```

Comment:
`read_registers` makes no assumptions about the size of registers (it's callback accepts a slice of u8s, which can be any size). This is deliberate, as some platforms have variable-sized registers (e.g: x86's floating point and vector extensions)

`write_registers` needs to provide the target with the raw `&[u8]` that was parsed from the packet (or potentially an iterator u8, if you want to be fancy), leaving the interpretation up to the individual targets.